### PR TITLE
big steps towards compatibility with python3

### PR DESCRIPTION
**Pull request by fchapoton**
I have made a lot of changes in order to make the code compatible with python3 and sage 9.0

### CODE REVIEW

**Comment by fchapoton** [comment]
ping ?

**Comment by djohnson2718** [comment]
Looks great! Sorry for the slow response. I hadn't realized that sage had switched to python 3 until I looked at this.

Does it work now, or is it still in progress?

Once it is working, we could put an updated version on pypi as well.

**Review by djohnson2718 — COMMENTED** [review]
looks good

File: strataalgebra/allstrata.py [hunk 1]
```diff
@@ -650,9 +653,9 @@ def decorate1_list(G,r):
             #print i, vec, vec[i] 
             S_list.append(Partitions(vec[i]))
         for i in range(a,a+b):
-            S_list.append([[vec[i]-j,j] for j in range(Rational((vec[i],2)) + 1)])
+            S_list.append([[vec[i]-j,j] for j in range(vec[i] // 2 + 1)])
```
Comment (djohnson2718) [inline strataalgebra/allstrata.py:656]: Is "//" correct here? I think in python3 "//" gives the floor after division, but we want rational division here.

Comment (fchapoton) [reply strataalgebra/allstrata.py:656]: Well, range now only accepts integer arguments, so there is no choice.

Comment (fchapoton) [reply strataalgebra/allstrata.py:656]: I get some failing doctests there:

sage -t strataalgebra/strataalgebra.py  # 3 doctests failed
sage -t topintersections/index.rst  # 1 doctest failed

but you have very few doctests globally..

Comment (djohnson2718) [reply strataalgebra/allstrata.py:656]: Ah, I didn't notice it was inside a range.

Comment (fchapoton) [reply strataalgebra/allstrata.py:656]: Coverage score is low:

```
:~/mgn$ sage -coverage --summary strataalgebra/
Global score: 12.0% (17 of 142)

6 files with wrong documentation
100 functions with no doc
25 functions with no test
0 doctest are potentially wrong

Files with wrong documentation:
-------------------------------
  strataalgebra/allstrata.py
  strataalgebra/intersect.py
  strataalgebra/strataalgebra.py
  strataalgebra/stratagraph.py
  strataalgebra/stratagraph2.py
  strataalgebra/tests.py
:~/mgn$ sage -coverage --summary topintersections/
Global score: 0.6% (1 of 165)

10 files with wrong documentation
84 functions with no doc
80 functions with no test
0 doctest are potentially wrong

Files with wrong documentation:
-------------------------------
  topintersections/TautRing3.py
  topintersections/checkin.py
  topintersections/exprwdata.py
  topintersections/intersection7.py
  topintersections/intersection_ui.py
  topintersections/push_down.py
  topintersections/remember.py
  topintersections/richcomparisonmixin.py
  topintersections/tau.py
  topintersections/testvsMgnLb.py

```

**Pull request merged**